revised README (#16)
Unity-Technologies/DOTS-training-samples: README.md

Under the 'Originals' directory, you'll find small simulations/games implemented in classic (non-DOTS) Unity. Your task is to implement one of these samples in DOTS.

1. Clone this repository.
2. Make a new branch for your sample port.
3. Create a new Unity Project and place it under the 'Ported' directory, *e.g/Ported/AutoFarmers/
4. Configure the project for DOTS

> Feel free to copy assets and code snippts from the orignials into your ports. The focus of this exercise is to familiarize yourself with the DOTS API and to practice data-oriented design principles!

Reviewer: ```suggestion
Feel free to copy assets and code snippets from the originals into your ports. The focus of this exercise is to familiarize yourself with the DOTS API and to practice data-oriented design principles!
```

<li>Workers in the 'pass full' line look for full buckets on the ground within a small radius. When found, they walk over, pick up the bucket, carry it to the next worker up the line, and drop it. The last worker in the 'pass full' line empties the bucket on the fire (lowering the temperature of that cell and its eight surrounding cells to zero). Workers move slower while carrying a full bucket.</li>
<li>Workers in the 'pass empty' line behave the same but only carry empty buckets. The last worker in the 'pass empty' line will fill the bucket in the nearest water.</li>
<li>Every frame, a team will reposition its line to the fire cell closest to its current water source.</li>
<li>When a team's water source is exhausted, it repositions the line to the next closest water.</li>
<li>When repositioning, a worker will not resume its normal behavior until it has reached its new position.</li>
<li>In the moment between a worker placing a bucket on the ground and the next picking it up, the line might reposition, in which case the bucket will be left on the ground.</li>

> <li>OPTIONAL: A configured small number of omniworkers pick up the nearest bucket, fill it at the nearest water source, and dump it on the nearest fire.</li>

Reviewer: I don't understand the term "omniworker", does that mean independent worker (without a team)?

> <li>OPTIONAL: A configured small number of omniworkers pick up the nearest bucket, fill it at the nearest water source, and dump it on the nearest fire.</li>

Author: ```suggestion
<li>OPTIONAL: A small configured number of 'omniworkers' (black) are not part of any team. An omniworker picks up the nearest bucket, fills it at the nearest water source, and dumps it on the nearest fire.</li>
```

<li>Keyboard controls allow the user to reset the simulation.</li>
   </ul>
</details>

![Bucket Brigade](_imgs/BucketBrigade.png?raw=true)

<details>
  <summary><strong>Combat Bees</strong>: Two bee hives battle for resources and survival.
  </summary>
    
   <ul>

> <li>Resources (green discs) spawn in the middle. Yellow bees spawn in the yellow base. Blue bees spawn in the blue base.</li>

Reviewer: ```suggestion
<li>Resources (green discs) spawn in the middle. Yellow bees spawn in the yellow base. Blue bees spawn in the blue base.</li>
<li>Resources spawned at the same location will stack on top of each other.</li>
```

<li>Bees pick up resources and drop them in their base.</li>
<li>When a resource hits the ground of a base, it explodes, spawning several bees of that base's color.</li>
<li>Bees not carrying resources may attack and destroy enemy bees carrying resources.</li>
<li>A destroyed bee emmits bee fragments and blood splatters, and the resource falls to the ground. Blood splatters on surfaces shrink to nothing over time.</li>
<li>Each bee's displayed scale oscilates along all three axes, but this is just a visual effect. Collisions are not affected.</li>

> <li>Keyboard controls allow the user to reset the simulation.</li>

Reviewer: ```suggestion
<li>Keyboard controls allow the user to reset the simulation.</li>
<li>Pointer clicks spawn additional resources at the point clicked (determined by casting a ray from the cursor).</li>
```

<li>Cars have three main states: cruising, looking to change lane, and overtaking.</li>
<li>While cruising, a car accelerates/brakes to its cruising speed; if it comes up behind a slower car within its overtake distance, it enters the 'looking to change lane' state.</li>
<li>A looking-to-change car will look for sufficient open space to its left or right to change lanes. While blocked from changing lanes, the car will tail the car ahead of it. When it is clear to change lane, it will enter the overtake state.</li>
<li>In the overtake state, a car first moves into another lane, then accelerates to its overtake speed. Once the car's overtake time elapses, it looks to move back into the lane it came from. If blocked from moving back, the car gives up and goes back to the cruising state. If at any time the car is impeded by a slower moving car within its overtake distance, it goes back to the cruising state.</li>
<li>Keyboard controls allow the user to reset the simulation.</li>
   </ul>
</details>

![Highway Racers](_imgs/HighwayRacers.gif?raw=true)

<details>
  <summary><strong>Jump The Gun</strong>: A ball jumps through a blocky landscape and avoids cannonballs.
  </summary>

<ul>

> <li>The ball bounces between columns to reach the column under the mouse cursor. (This requires computing the appropriate trajectory for each bounce.)</li>

Reviewer: ```suggestion
<li>The ball bounces from column to adjacent column towards the mouse cursor. (This requires computing the appropriate trajectory for each bounce.) The movement is clamped to the edges of the playing field.</li>
```

<li>On init, the cannons spawn on random columns. The cannons always turn to face the player's ball.</li>
<li>Periodically, each cannon fires a cannon ball along a trajectory that will intersect the player's ball (at its current position) but not hit any columns in between.</li>

> <li>When a cannon ball hits the top of a column, the impact pushes the column down.</li>

Reviewer: ```suggestion
<li>When a cannon ball hits the top of a column, the impact pushes the column down (but not below the minimum height).</li>
```

<li>The game is over when a cannon ball hits the player's ball.</li>
<li>Keyboard controls allow the user to reset the simulation.</li>
   </ul>  

</details>

![Jump The Gun](_imgs/JumpTheGun.gif?raw=true)

<details>
  <summary><strong>LabRat</strong>: Place arrows on the board to route the most rats into your home base.
  </summary>
    
   <ul>

> <li></li>

Reviewer: ```suggestion
```

> <li>Mice spawn at frequent, randomized intervals in the top and bottom corners. The top mice head down. The bottom mice  head up.</li>

Reviewer: ```suggestion
<li>Mice spawn at frequent, randomized intervals in the top and bottom corners. The top mice head down. The bottom mice head up.</li>
```

<li>At start of play, walls are randomly placed between grid cells.</li>
<li>Cats and mice travel on the grid, changing direction when they hit a wall or travel over an arrow.</li>
<li>Cats spawn in random squares. When a cat and mouse intersect, the mouse is eaten.</li>

> <li>When a mouse hits a player's 'home base' (one of the four dots placed near the center of the grid), the mouse disappers, and the player is awarded a point.</li>

Reviewer: ```suggestion
<li>When a mouse hits a player's 'home base' (one of the four dots placed near the center of the grid), the mouse disappears, and the player is awarded a point.</li>
```

<li>Players can place arrows in cells of the board. The green player places green arrows, the red player places red arrows, etc. A player cannot place their arrows in a cell occupied by an arrow of another player.</li>
<li>Once a player has three arrows on the board, their next placed arrow removes their oldest arrow on the board.</li>
<li>Only one player is human. The AI players just place their arrows randomly at random intervals.</li>
<li>At the end of 30 seconds, the player with the most points wins.</li>
<li>Keyboard controls allow the user to reset the simulation.</li>
   </ul>
</details>

![Lab Rat](_imgs/LabRat.gif?raw=true)

<details>
  <summary><strong>Magnetic Roads</strong>: Cars drive along 3D generated splines in all orientations.
  </summary>
    
   <ul>

> <li>Cars drive in two lanes on both sides of the rode. Cars always drive in the right lane.</li>

Reviewer: ```suggestion
<li>Cars drive in two lanes on both sides of the road. Cars always drive in the right lane.</li>
```

<li>At three-way intersections, each car randomly chooses whether to go left, right, or straight.</li>
<li>Cars wait to enter an intersection if their path through the intersection crosses the path of another car in the intersection.</li>
<li>Keyboard controls allow the user to reset the simulation.</li>
   </ul>
</details>

![Magnetic Roads](_imgs/MagneticRoads.gif?raw=true)   

<details>
  <summary><strong>Metro</strong>: Metro train transport commuters from station to station.
  </summary>
    
   <ul>
   <li>The trains move on a fixed schedule. Passengers should not enter or exit the train if they do not have time to get in position before the train moves.</li>
   <li>Disembarking passengers randomly pick another train to board.</li>
   <li>On init, the tracks and station placement is randomly generated.</li>
   <li>Keyboard controls allow the user to reset the simulation.</li>
   </ul>
</details>

![Metro](_imgs/Metro.gif?raw=true)
 
<details>
  <summary><strong>Stack Interchange</strong>: Cars drive through a stack interchange.
  </summary>
   
   <ul>

>    <li>At random intervals, cars spawn in at the edge of the straight roads. Cars despawn when they run off the edge of the straight roads.</li>

Reviewer: ```suggestion
   <li>At random intervals, cars spawn at the edge of the straight roads. Cars despawn when they run off the edge of the straight roads.</li>
```
